lib/scholar/impute/simple_imputer.ex

```diff
@@ -0,0 +1,242 @@
+  defstruct [:statistics, :missing_values]
+
+  opts_schema = [
+    missing_values: [
+      type: {:or, [:float, :integer, {:in, [:nan]}]},
+      default: :nan,
+      doc: ~S"""
+      The placeholder for the missing values. All occurrences of missing_values will be imputed.
+      """
+    ],
+    strategy: [
+      type: {:in, [:mean, :median, :mode, :constant]},
+      default: :mean,
+      doc: ~S"""
+      The imputation strategy.
+
+      * `:mean` - replace missing values using the mean along each column.
+
+      * `:median` - replace missing values using the median along each column.
+
+      * `:mode` - replace missing using the most frequent value along each column.
+        If there is more than one such value, only the smallest is returned.
+
+      * `:constant` - replace missing values with `:fill_value`.
+      """
+    ],
+    fill_value: [
+      type: {:or, [:float, :integer]},
+      default: 0.0,
+      doc: ~S"""
+      When strategy is set to `:constant`, `:fill_value` is used to replace all occurrences of missing_values.
+      """
+    ]
+  ]
+
+  @opts_schema NimbleOptions.new!(opts_schema)
+
+  @doc """
+  Univariate imputer for completing missing values with simple strategies.
+
+  ## Options
+
+  #{NimbleOptions.docs(@opts_schema)}
+
+  ## Returns
+
+    The function returns a struct with the following parameters:
+
+    * `:missing_values` - the same value as in `:missing_values`
+
+    * `:statistics` - The imputation fill value for each feature. Computing statistics can result in `Nx.Constant.nan/1` values.
+
+  ## Examples
+
+    iex> x = Nx.tensor([[1, 2, :nan], [3, 7, :nan], [:nan, 4, 5]])
+    iex> imputer = Scholar.Impute.SimpleImputer.fit(x, strategy: :mean)
+    iex> Scholar.Impute.SimpleImputer.transform(imputer, x)
+    #Nx.Tensor<
+      f32[3][3]
+      [
+        [1.0, 2.0, 5.0],
+        [3.0, 7.0, 5.0],
+        [2.0, 4.0, 5.0]
+      ]
+    >
+
+    iex> x = Nx.tensor([[1, 2, :nan], [3, 7, :nan], [:nan, 4, 5]])
+    iex> y = Nx.tensor([[7, :nan, 6], [6, 9, :nan], [8, :nan, 1]])
+    iex> imputer = Scholar.Impute.SimpleImputer.fit(x, strategy: :median)
+    iex> Scholar.Impute.SimpleImputer.transform(imputer, y)
+    #Nx.Tensor<
+      f32[3][3]
+      [
+        [7.0, 4.0, 6.0],
+        [6.0, 9.0, 5.0],
+        [8.0, 4.0, 1.0]
+      ]
+    >
+
+  """
+  deftransform fit(x, opts \\ []) do
+    validated_opts = NimbleOptions.validate!(opts, @opts_schema)
+
+    input_rank = Nx.rank(x)
+
+    if input_rank != 2 do
+      raise ArgumentError, "Wrong input rank. Expected: 2, got: #{inspect(input_rank)}"
+    end
+
+    if validated_opts[:missing_values] != :nan and
+         Nx.any(Nx.is_nan(x)) == Nx.tensor(1, type: :u8) do
+      raise ArgumentError,
+            ":missing_values other than :nan possible only if there is no Nx.Constant.nan() in the array"
+    end
+
+    {type, _num_bits} = Nx.type(x)
+
+    x =
+      if validated_opts[:strategy] == :constant do
```

Comment:
Perhaps you could remove the if by adding this as a condition to the first 2 cond clauses